packages/coinstac-simulator/src/db-server.js

```diff
@@ -5,58 +5,126 @@
  * @module db-server
  */
 
-require('./handle-errors')();
-const pdbs = require('spawn-pouchdb-server');
-const cp = require('child_process');
-const pdbsConfig = require('./.pouchdb-server-config');
+require('./utils/handle-errors');
+
 const cloneDeep = require('lodash/cloneDeep');
+const common = require('coinstac-common');
+const config = require('./utils/config');
+const logger = require('./utils/logger');
+const pdbs = require('spawn-pouchdb-server');
+const Pouchy = require('pouchy');
+const url = require('url');
+
+const Consortium = common.models.Consortium;
+
+/**
+ * Placeholder for the server.
+ *
+ * @type {(undefined|Function)}
+ */
+let server;
 
-const me = {};
+function addDocument(pathname, doc) {
+  const dbUrl = url.format({
+    hostname: 'localhost',
+    pathname,
+    port: config.port,
+    protocol: 'http',
+  });
+
+  const db = new Pouchy({
+    url: dbUrl,
+    sync: 'out',
+  });
+
+  return db.save(doc);
+}
 
 /**
  * @function setup
  * @description boots a pouchdb-server, a dbRegistry instance, and
  * a computation registry instance.  these utilities are commonly
  * required for PipelineRunnerPool testing
+ * @param {string} declPath Path to the declaration file
  * @returns {Promise}
  */
-const setup = () => { // eslint-disable-line
-  return new Promise((res, rej) => {
-    try { cp.execSync('pkill -f pouchdb-server'); } catch (err) {} // eslint-disable-line
-    // @note ^^^ helpful to uncomment when tests are failing
+function setup(declPath) {
+  if (server) {
+    return Promise.reject(new Error('Server already instantiated'));
+  }
+
+  const decl = require(declPath); // eslint-disable-line global-require
 
+  return new Promise((resolve, reject) => {
     // spawn-pouchdb-server mutates user input >:(
     // https://github.com/hoodiehq/spawn-pouchdb-server/pull/33
-    pdbs(cloneDeep(pdbsConfig), (err, srv) => {
-      if (err) { return rej(err); }
-      me.server = srv;
-      return res(srv);
+    pdbs(cloneDeep(config), (error, srv) => {
+      if (error) {
+        reject(error);
+      } else {
+        server = srv;
+        resolve(server);
+      }
     });
-  });
-};
+  })
+    .then(() => {
```

Comment:
Looks fine to me. Without confusing diff: https://github.com/MRN-Code/coinstac/blob/934ad12193ae53a6cd3c376a264bffefa8db73dc/packages/coinstac-simulator/src/db-server.js#L70